### PR TITLE
Note that `-p` is also necessary

### PR DESCRIPTION
**Pull request by jglick**
Otherwise the Docker container will not expose its port 22 from the host and you will not be able to connect to it. Am I missing something?

### CODE REVIEW

**Comment by dduportal** [comment]
Cc @gounthar for info and help.

we discussed improving this doc because there are 2 main use cases here (which are half/half) so we want to describe both.

- if the jenkins controller is executed in a container in the same container engine, then no port publication/exposition should be done at all. The image already specifies that its port 22 should be exposed in the private networks so Docker knows what to do.
- If the jenkins controller is running « outside » a container, or if the end user want to ssh-connect to the agent themselves, then yes this should be added to « publish » the port 22 of the private network to another port on the host


about the 2nd case, a few suggestions:
- we have to use the « publish » wording, not the « expose » wording (linking to docker documentation could help clarify)
- Using long form of the « -p » flag (eg « —publish ») would also help clarifying
- Using a port different than 22 (2222?) on the host would also help : not only it is less probable to be already used on the host system, but it also helps users to « map » in their mind what port is what (localhost:2222 vs container:22)

**Comment by jglick** [comment]
> if the jenkins controller is executed in a container in the same container engine

You mean if you also ran `docker run jenkins/jenkins` and referred to the agent container as `agent:22`? Does that even work these days without Docker Compose?

> a few suggestions

Agreed with all.

**Comment by dduportal** [comment]
> > if the jenkins controller is executed in a container in the same container engine
> 
> You mean if you also ran `docker run jenkins/jenkins` and referred to the agent container as `agent:22`? Does that even work these days without Docker Compose?

Good point: docker-compose does define [user-defined networks with DNS resolution](https://docs.docker.com/network/bridge/#manage-a-user-defined-bridge) so it works if you use `agent:22`.

Otherwise, with a `docker run` command which does not specify any `--network` flag, then you must use the private IP of the agent container that must be in the same "default bridge network" (ref. https://docs.docker.com/network/bridge/#use-the-default-bridge-network).
For the "default bridge" we must NOT mention the `--link` flag because it is deprecated (ref. https://docs.docker.com/network/links/)  even though it is still working as for the 20.10 version of Docker CE.

Ref. https://docs.docker.com/network/bridge/#differences-between-user-defined-bridges-and-the-default-bridge.

=> It might be important to write a note or at least link. The `docker run` command examples that we would provide should uses user-defined networks (or compose). WDYT?


=> Side note: `docker compose` is part of `docker` (https://docs.docker.com/compose/reference/) since a few months: we can rely on Docker Compose being available.

**Comment by jglick** [comment]
> we can rely on Docker Compose being available

Sure, my point is just that the current instructions make no mention of Compose, or of the controller being managed in a container as well, they just present a `docker run` command to create an agent container without any context and that command will not in fact work unless you have set up a bunch of other stuff in the background.

**Comment by dduportal** [comment]
> > we can rely on Docker Compose being available
> 
> Sure, my point is just that the current instructions make no mention of Compose, or of the controller being managed in a container as well, they just present a `docker run` command to create an agent container without any context and that command will not in fact work unless you have set up a bunch of other stuff in the background.

Make sense: could you only use the "long" form so we can proceed merging this PR (see suggestion)? 

My proposals are object of subsequent PR and should not block yours here in any way

File: README.md [hunk 1]
```diff
@@ -17,12 +17,13 @@ See [Jenkins Distributed builds](https://wiki.jenkins-ci.org/display/JENKINS/Dis
 To run a Docker container
 
 ```bash
-docker run -d --rm --name=agent -e "JENKINS_AGENT_SSH_PUBKEY=<public_key>" jenkins/ssh-agent
+docker run -d --rm --name=agent -p 22:22 -e "JENKINS_AGENT_SSH_PUBKEY=<public_key>" jenkins/ssh-agent
```
Comment (dduportal) [inline README.md:20]:
```suggestion
docker run -d --rm --name=agent --publish 2200:22 -e "JENKINS_AGENT_SSH_PUBKEY=<public_key>" jenkins/ssh-agent
```

**Review by dduportal — APPROVED** [review]

**Review by dduportal — APPROVED** [review]
many thanks!

**Pull request merged**